packages/cli/source/manifest/compiler.ts

```diff
@@ -242,61 +133,93 @@ const compile = async (): Promise<Manifest> => {
         return manifest;
     });
 
-    let app: App | null = null;
-    let queries: {
-        [key: string]: Query;
-    } = {};
-    let resources: {
-        [key: string]: Resource;
-    } = {};
+    let manifestResult: {
+        app: object;
+        queries: object;
+        resources: object;
+        values: object;
+        file: string;
+    } = {
+        app: {},
+        queries: {},
+        resources: {},
+        values: {},
+        file: "",
+    };
+
+    let queries: Query[] = [];
+    let resources: Resource[] = [];
+
     for (const manifest of manifests) {
         for (const key in manifest) {
             switch (key) {
                 case "app": {
-                    if (app) {
+                    if (!lodash.isEmpty(manifestResult.app)) {
                         logDuplicateError("app", manifest.file);
                     }
-                    app = parseApp(manifest.app, manifest.file);
+
+                    const { error, value } = appSchema.validate(manifest.app, {
+                        stripUnknown: true,
+                    });
+
+                    if (error) {
+                        logSemanticError(error.message);
+                        break;
+                    }
```

Comment:
Why is this necessary? We are doing the same thing, but with the final merged object in line 213.